### ISSUE TITLE
Release-please opened 2 PRs with the same version

### ISSUE THREAD

**Issue by parthea** [googleapis/release-please#826]
Release please is lightning fast at opening PRs! I have a strange issue where release please opened 2 PRs in [python-gke-hub](https://github.com/googleapis/python-gke-hub/pulls?q=is%3Apr+is%3Aclosed) with the same version. Somehow I managed to merge both PRs :man_shrugging: . Anyway, I noticed that both PRs had `autorelease: pending` tags set so there could be a race condition where a new release PR could be opened with a same version while autorelease is running. I'm marking this issue as a feature request since it is rare.

**Comment by bcoe** [comment]
@parthea thank you 👌 labeled this as a bug, as I'm pretty sure it's a regression.

**Comment by chingor13** [comment]
Another example:

1. https://github.com/googleapis/nodejs-retail/pull/70 was opened at 2:48pm, merged at 3:36pm.
2. https://github.com/googleapis/nodejs-retail/pull/71 was opened at 3:36 triggered by the push to master from PR 70 being merged

**Comment by chingor13** [comment]
Under the hood, release-please looks to see if there's any pending merged pull requests. This uses a GraphQL query to find commits to the release branch and their associated pull requests. It looks like there's a race condition where the GraphQL query does not return the commit/associated pull request for the newly merged release PR.

**Comment by chingor13** [comment]
A few options:

1. introduce a small delay (eww)
2. if we fail to find one, run the same logic again to be sure
3. add a fallback to check the just the latest commit somehow

**Comment by chingor13** [comment]
Logs:
The merge of PR 70 triggered a push event at 22:36:07
release-please queried GraphQL somewhere between 22:36:09 and 22:36:13
The new pull request was created at 22:36:17
```
I 2021-07-13T22:36:07.443Z release_please scheduling cloud task targetting: functions release_please ats9qjcol8zz  nodejs-retail GitHub Webhook push tmatsuo
I 2021-07-13T22:36:07.443Z release_please scheduling task in queue release-please release_please ats9qjcol8zz  nodejs-retail GitHub Webhook push tmatsuo
I 2021-07-13T22:36:07.443Z release_please uploading payload to 1626215767443-bbf0ad8a-7fbd-4be4-8a32-86a397dce0a7.txt release_please ats9qjcol8zz  nodejs-retail GitHub Webhook push tmatsuo
I 2021-07-13T22:36:07.848Z release_please ats9psf0vvmm nodejs-retail GitHub Webhook push tmatsuo signature = sha1=redacted release_please ats9psf0vvmm  nodejs-retail GitHub Webhook push tmatsuo
I 2021-07-13T22:36:08.847Z release_please ats9tukh2acl Cloud Task downloaded payload from 1626215767443-bbf0ad8a-7fbd-4be4-8a32-86a397dce0a7.txt release_please ats9tukh2acl   Cloud Task  
I 2021-07-13T22:36:09.165Z release_please ats9tukh2acl Cloud Task push (googleapis/nodejs-retail) release_please ats9tukh2acl   Cloud Task  
I 2021-07-13T22:36:13.544Z release_please ats9tukh2acl Cloud Task found 3 commits since 068df3183df699142fb4e7e4245ae78f04fecc9f release_please ats9tukh2acl   Cloud Task  
I 2021-07-13T22:36:13.560Z release_please ats9tukh2acl Cloud Task release as patch: There are 0 BREAKING CHANGES and 0 features release_please ats9tukh2acl   Cloud Task  
W 2021-07-13T22:36:13.971Z release_please ats9tukh2acl Cloud Task file package-lock.json did not exist release_please ats9tukh2acl   Cloud Task  
W 2021-07-13T22:36:14.126Z release_please ats9tukh2acl Cloud Task file npm-shrinkwrap.json did not exist release_please ats9tukh2acl   Cloud Task  
I 2021-07-13T22:36:14.293Z release_please ats9tukh2acl Cloud Task updating @google-cloud/retail dependency in samples/package.json from ^1.1.5 to ^1.1.5 release_please ats9tukh2acl   Cloud Task  
I 2021-07-13T22:36:14.456Z release_please ats9tukh2acl Cloud Task updating CHANGELOG.md release_please ats9tukh2acl   Cloud Task  
I 2021-07-13T22:36:14.456Z release_please ats9tukh2acl Cloud Task updating package.json from 1.1.5 to 1.1.5 release_please ats9tukh2acl   Cloud Task  
I 2021-07-13T22:36:14.457Z release_please ats9tukh2acl Cloud Task Starting GitHub PR workflow... release_please ats9tukh2acl   Cloud Task  
I 2021-07-13T22:36:14.624Z release_please ats9tukh2acl Cloud Task Successfully found branch HEAD sha "3954a6ae387b3659e6b188473055a25ff52078d3". release_please ats9tukh2acl   Cloud Task  
I 2021-07-13T22:36:15.140Z release_please ats9tukh2acl Cloud Task Successfully created branch at https://api.github.com/repos/googleapis/nodejs-retail/git/refs/heads/release-v1.1.5 release_please ats9tukh2acl   Cloud Task  
I 2021-07-13T22:36:15.317Z release_please ats9tukh2acl Cloud Task Got the latest commit tree release_please ats9tukh2acl   Cloud Task  
I 2021-07-13T22:36:15.676Z release_please ats9tukh2acl Cloud Task Successfully created a tree with the desired changes with SHA 0a86e0d97f4b4d844a6b05eac05ce6a61277c4e8 release_please ats9tukh2acl   Cloud Task  
I 2021-07-13T22:36:16.064Z release_please ats9tukh2acl Cloud Task Successfully created commit. See commit at https://api.github.com/repos/googleapis/nodejs-retail/git/commits/773147ca807d451b5541ce20ac6bd64d977498ac release_please ats9tukh2acl   Cloud Task  
I 2021-07-13T22:36:16.540Z release_please ats9tukh2acl Cloud Task Successfully updated reference release-v1.1.5 to 773147ca807d451b5541ce20ac6bd64d977498ac release_please ats9tukh2acl   Cloud Task  
I 2021-07-13T22:36:17.840Z release_please ats9tukh2acl Cloud Task Successfully opened pull request available at url: https://api.github.com/repos/googleapis/nodejs-retail/pulls/71. release_please ats9tukh2acl   Cloud Task  
I 2021-07-13T22:36:17.840Z release_please ats9tukh2acl Cloud Task Successfully opened pull request: 71. release_please ats9tukh2acl   Cloud Task  
I 2021-07-13T22:36:17.840Z release_please ats9tukh2acl Cloud Task adding label autorelease: pending to https://github.com/googleapis/nodejs-retail/pull/71 release_please ats9tukh2acl   Cloud Task  
I 2021-07-13T22:36:18.241Z release_please ats9tukh2acl Cloud Task find stale PRs with label "autorelease: pending" release_please ats9tukh2acl   Cloud Task  
I 2021-07-13T22:36:18.841Z release_please ats975b7nsfe Cloud Task signature = sha1=redacted release_please ats975b7nsfe   Cloud Task  
```

**Comment by marcopeise** [comment]
#826 fixed retry logic for **finding** merged release PRs when GraphQL indexing lags. A related gap remains: **adding** labels immediately after `createPullRequest` still has no retry — `addIssueLabels` can fail with 422 `Issue not found` (GitHub eventual consistency).

Tracked in #2754. Fix PR: #2782.
